xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -514,11 +500,11 @@ public void onClose(Status status, Metadata trailers) {
         }
       }
 
-      filterInterceptors.add(new ClusterSelectionInterceptor());
       return
           Result.newBuilder()
               .setConfig(config)
-              .setInterceptor(combineInterceptors(filterInterceptors))
+              .setInterceptor(combineInterceptors(
+                  ImmutableList.of(filters, new ClusterSelectionInterceptor())))
```

Comment:
`filters` may be null here, so ImmutableList will throw NPE.  Should we check for this?

Reply:
How can `filters` be null? It has checkNotNull() in RouteData, and we can't get here if cluster is still null (we'd still be in the first iteration of the do-while).

Reply:
You may be right.

Just from the quick glance at the code I've noticed it's initiated as `null`, and then set in `if - elseif - elseif` chain; without final `else`. Then after the while loop the only exit condition is `config == null`.

But looks like it's always set next to `cluster`, which presence seems to guaranteed by `retainCluster`.

Reply:
Also the previous code didn't have this issue because of `filterInterceptors` was an array list (which also was guaranteed to be non-empty by `filterInterceptors.add(new ClusterSelectionInterceptor());`.

This was enough signal for me to flag a potential NPE in the review.